mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -418,7 +418,17 @@ impl Nexus {
             nexus_target: None,
         });
 
-        n.bdev.set_uuid(uuid.map(String::from));
+        if let Some(s) = uuid {
+            match uuid::Uuid::parse_str(s) {
+                Ok(u) => {
+                    info!("nexus {}: setting bdev UUID: {}", name, s);
+                    n.bdev.set_uuid(u);
+                }
+                Err(error) => {
+                    warn!("nexus {}: invalid UUID: {}: {}", name, s, error);
```

Comment:
Seems this should reutrn the error and destruct the nexus.

Reply:
This is consistent with the previous behaviour which was to generate a new random UUID. This is unnecessary as the bdev already has a valid UUID (randomly generated at the point of creation).

Reply:
This might be a bug. Otherwise it'd mean the nexus gets shared via nvmf on the wrong nqn? The csi will not be able to connect to it?

Reply:
I'm coming around to the view that we probably ought to fail outright in this case.

Reply:
Passing the error up would require more changes that I'd rather not make at this time. Have now changed code to effectively replicate the old behaviour: in the case that no uuid is specified, or if the specified uuid cannot be parsed, then auto-generate a new one.